### DISCUSSION TITLE
Unknown arg `path` in prisma type JsonFilter

### DISCUSSION THREAD

**Discussion by Vitaee** [prisma/prisma#10040] [Q&A] [answered]
Hi everyone,

I am learning prisma and I'm facing an error that I can't find the reason for. I have basic nodejs project which I created from [there.](https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/relational-databases-node-postgres)

Here is my model;
```prisma
model Boat {
  id     Int     @id @default(autoincrement())
  boatjs Json 
  createdAt DateTime @default(now())
  addedBy String @default("Can")
}
```

I am using postgreSQL and I want to use Json field to save or filter the json object which is named as ```boatjs```.
Here is a simple json data;
```js
{
    "url": "https://www.yachtworld.com/yacht/2014-beneteau-oceanis-41-8077182/",
      "id": 8077182,
      "title": "2014 Beneteau Oceanis 41",
      "fetchTimeStamp": 1634900944013,
      "fetchDate": "Friday, October 22, 2021",
      "listingDate": {
        "created": "2021-10-21T11:52:28Z",
        "modified": "2021-10-21T12:45:36Z",
        "sold": null
      },
      "price": 165897.72,
      "currency": "USD",
      "isTaxPaid": null,
      "make": "Beneteau",
      "model": "Oceanis 41",
      "yearBuilt": 2014,
      "length": 41,
      "unitOfLength": "ft",
      "engine": null,
      "fuel": "diesel",
      "location": {
        "country": "HR",
        "city": "Upon request",
        "subdivision": null
      }
  }
  ```
  
  I can save this object there is a no problem. But when I want to do filter operations on json object. I am facing an issue which is;
  
  ```
  Unknown arg `path` in where.boatjs.path for type JsonFilter. Did you mean `not`? Available args:
type JsonFilter {
  equals?: JsonNullValueFilter | Json
  not?: JsonNullValueFilter | Json
}
```
Here is my function to get json object;

```
const models = await prisma.boat.findMany({
        where: { 
            boatjs: {
                path: ['model'], 
                equals: 'Oceanis 41'
        },
    },
    });
    
    res.json({
        success: true,
        payload: models,
    })
```

But its not working and I can't find any helpful resources about this issue. I don't think this error is caused by me. Here is info about node version etc.

```
node -v === v16.13.0
```

```
prisma --version
prisma                  : 3.3.0
@prisma/client          : 3.3.0
Current platform        : debian-openssl-1.1.x
```

I am using ``` Ubuntu 20.04.3 LTS ```. 

Finally,
I tried to fix this issue doing like re-install the prisma or changing node version or creating another project on another PC But still I am facing this issue,  when I want to do filter operations for json object.

Yours truly.

**Comment by austincrim** [comment]
Hi @Vitaee!

Have you enabled the `filterJson` preview feature in your `schema.prisma`? It looks something like this:
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["filterJson"]
}
```
If you don't have that, could you try adding it, rerunning `prisma generate`, and seeing if that resolves your issue?

**Reply by Vitaee** [reply]
Hi @austincrim!

Thanks for reply, I followed your suggestions then error is gone. 

Firstly, I add ```  previewFeatures = ["filterJson"] ``` to ```schema.prisma``` then I run ```prisma generate``` and its fixed the error. But know I am facing another issue let me write it. 

In my table I have json object for example;

```
[
  {
    "id": 8076466,
    "url": "https://www.yachtworld.com/yacht/2004-dufour-dufour-40-performace-8076466/",
    "fuel": "diesel",
    "make": "Dufour",
    "model": "DUFOUR 40 PERFORMACE",
    "price": 104200.62,
    "title": "2004 Dufour DUFOUR 40 PERFORMACE",
    "engine": "VOLVO",
    }
 ]
 ````
 
 and I am doing simple filtering like;
 
 ```
 const data = await prisma.boat.findMany({
  where:{
    boatjs: {
      path: ["engine"],
      equals: "VOLVO"
    },
  },
});

 console.log(data);
 ```
 Then the query returned ``` [ ] ``` an empty list :(  In addition when I did filter by ``` Id ``` I get whole row there is a no problem. But I wan to do filter operations on ``` boatjs ``` variable which is the json object.. Shall I create another new project ? 
 
 In summary, for now when I want to use ``` path : [] ``` variable for filtering, its returns ``` [ ] ```. Even the data exist in postgreSQL db. 
 
 Thanks.

**Comment by Vitaee** [comment, marked as answer]
I fix the issue! :D 

Here is my mistake; I was saving the json data like this;

```
let js_data = [
  {
    "id": 8076466,
    "url": "https://www.yachtworld.com/yacht/2004-dufour-dufour-40-performace-8076466/",
    "fuel": "diesel",
    "make": "Dufour",
    "model": "DUFOUR 40 PERFORMACE",
    "price": 104200.62,
    "title": "2004 Dufour DUFOUR 40 PERFORMACE",
    "engine": "VOLVO",
    }
 ]
 
 ####
 
 await prisma.boat.create({
 data:{
   boatjs: js_data
 },
});
 ```
 
 Then when I want to do filter operations I faced with ``` [] ``` list. Then I tried this;
 
 ```
 let js_data = 
 {
    "id": 8076466,
    "url": "https://www.yachtworld.com/yacht/2004-dufour-dufour-40-performace-8076466/",
    "fuel": "diesel",
    "make": "Dufour",
    "model": "DUFOUR 40 PERFORMACE",
    "price": 104200.62,
    "title": "2004 Dufour DUFOUR 40 PERFORMACE",
    "engine": "VOLVO",
    }
    
    await prisma.boat.create({
         data: {
                  boatjs: js_data
                  }               
    });
 
```
And filter operations are working perfectly and ofcourse I added ```  previewFeatures = ["filterJson"] ``` to schema.prisma file. So the mistake saving json object with list parentheses.. or is it mistake? Why we are facing this issue when we saving json object with ``` [ ] ``` ?


Yours truly

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
